jaxrs-api/src/main/java/javax/ws/rs/core/CacheControl.java

```diff
@@ -83,7 +87,10 @@ public CacheControl() {
      *
      * @throws IllegalArgumentException if the supplied string cannot be parsed
      *                                  or is null
+     * @deprecated This method will be removed in version 3.0. Please use
+     *   RuntimeDelegate.getInstance().createHeaderDelegate(CacheControl.class).fromString(value) instead.
      */
+    @Deprecated
```

Comment:
@NiklasMehner While I do support the removal of ```HEADER_DELEGATE``` I wonder why you want to get rid of ```valueOf``` and ```toString```?

Reply:
They only complicate things: 
* I don't see a use-case for an end-user of the jax-rs api
* For a jax rs implementation this is a case of the implementation calling an api method that does nothing but look up an implementation class to do the work. Why tunnel these calls through the api? It does not add any functionality but only complicates things (also the implementations I looked at will not call these methods).

Reply:
Getting rid of methods is a breaking change and definitively needs to go into 3.0, while getting rid of internal delegates does not break anything and can be fixed by implementors by simply overwriting the methods. Hence it is essential to understand your drivers.
* Getting rid of useless methods is something else than implementing them in a default way (which is non-breaking).
* There is a difference between getting rid of the delegation, and getting rid of the methods -even without delegation- at all. So you might want to split the PR?

Reply:
This PR is not breaking since it does not yet change anything but only deprecates the methods so the breaking change can be done in 3.0. 

I agree that removing the delegation is a different issue and can be done in a separate PR. But since this change is not included in this PR this would be an additional PR and not a split.

Overwriting the methods by an implementor is not possible, since the classes will be instantiated by an end-user of the api. An implementor has no chance to use an extended Version of
e.g. CacheControl:  Response.ok().cacheControl(new CacheControl());

Reply:
Agreed. :-)